plugins/eos/gs-plugin-eos.c

```diff
@@ -197,22 +198,35 @@ get_personality (void)
 	return g_strdup (personality);
 }
 
+static char *
+get_os_version_id (GError **error)
+{
+	g_autoptr(GsOsRelease) os_release = gs_os_release_new (error);
+
+	if (!os_release)
+		return NULL;
+
+	return g_strdup (gs_os_release_get_version_id (os_release));
+}
+
 gboolean
 gs_plugin_setup (GsPlugin *plugin,
 		 GCancellable *cancellable,
 		 GError **error)
 {
-	g_autoptr(GError) local_error = NULL;
 	GApplication *app = g_application_get_default ();
 	GsPluginData *priv = gs_plugin_get_data (plugin);
 
 	priv->session_bus = g_application_get_dbus_connection (app);
 
-	priv->desktop_apps = get_applications_with_shortcuts (plugin, cancellable,
-							      &local_error);
-	if (priv->desktop_apps == NULL)
-		g_warning ("Couldn't get the apps with shortcuts: %s",
-			   local_error->message);
+	{
+		g_autoptr(GError) local_error = NULL;
+		priv->desktop_apps = get_applications_with_shortcuts (plugin, cancellable,
+								      &local_error);
+		if (priv->desktop_apps == NULL)
+			g_warning ("Couldn't get the apps with shortcuts: %s",
+				   local_error->message);
+	}
```

Comment:
It does. Because it is when we'd start reusing the `local_error` otherwise. And I preferred to use scopes than clearing the error every time before the use (as I think it's easier to get it wrong with the latter).

Reply:
Fair enough.